Tests/DependencyInjection/Compiler/AnnotationCompilerPassTest.php

```diff
@@ -90,6 +90,21 @@ public function testMinimalAdmin()
         );
     }
 
+    public function testIdForAdmin()
+    {
+        /*
+         * @Admin(class="Sonata\AdminBundle\Entity\Foo", id="my.id")
+         */
+        $annotation = new Admin();
+        $annotation->class = 'Sonata\AdminBundle\Entity\Foo';
+
+        $meta = new ClassMetadata('Sonata\AdminBundle\Tests\Fixtures\Entity\Foo');
+
+        $annotation->processMetadata($meta);
+
+        $this->assertSame($meta->id == 'my.id');
```

Comment:
@wodka Workaround: `$ composer require phpspec/prophecy`. Don't forget to get rid of the `composer.json` changes.